assets/tests/components/app.test.tsx

```diff
@@ -1,10 +1,87 @@
 import React from "react"
-import renderer from "react-test-renderer"
-import App from "../../src/components/app"
+import renderer, { act } from "react-test-renderer"
+import * as Api from "../../src/api"
+import App, {
+  useFetchRoutes,
+  useFetchTimepoints,
+} from "../../src/components/app"
+import { setLoadingTimepointsForRoute } from "../../src/state"
+import { testHook } from "../testHelpers/testHook"
+
+// tslint:disable: react-hooks-nesting no-empty
+
+jest.mock("../../src/api", () => ({
+  __esModule: true,
+  fetchRoutes: jest.fn(() => new Promise(() => {})),
+  fetchTimepointsForRoute: jest.fn(() => new Promise(() => {})),
+}))
 
 test("renders", () => {
   const tree = renderer.create(<App />).toJSON()
-
   expect(tree).toMatchSnapshot()
+  // Flush effects
+  act(() => {})
 })
 
+describe("useFetchRoutes", () => {
+  test("fetches routes", () => {
+    const mockFetchRoutes: jest.Mock = Api.fetchRoutes as jest.Mock
+    act(() => {
+      testHook(() => {
+        useFetchRoutes(jest.fn())
+      })
+    })
+    expect(mockFetchRoutes).toHaveBeenCalledTimes(1)
+  })
+
+  test("fetches routes exactly once", () => {
```

Comment:
What about adding some context about when this might happen or what we are trying to avoid, e.g. something along the lines of, "doesn't re-fetch routes on subsequent renders"?